internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -877,8 +883,11 @@ func (r *ClusterDeploymentsReconciler) updateIfNeeded(ctx context.Context,
 	updateString(spec.ClusterName, cluster.Name, &params.Name)
 	updateString(spec.BaseDomain, cluster.BaseDNSDomain, &params.BaseDNSDomain)
 
-	update = r.updateNetworkParams(clusterDeployment, clusterInstall, cluster, params) || update
-
+	shouldUpdateNetworkParams, err := r.updateNetworkParams(clusterDeployment, clusterInstall, cluster, params)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to parse cluster OpenShift version")
```

Comment:
This error seems overly specific and includes information that I would expect to be present in the error being wrapped, the wrapping error should instead provide context on what was being done while the underlying error occurred, in this case something like "failed to update network params"